airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams/base_streams.py

```diff
@@ -78,9 +78,12 @@ def failure(response: FacebookResponse, request: Optional[FacebookRequest] = Non
             # although it is Optional in the signature for compatibility, we need it always
             assert request, "Missing a request object"
             resp_body = response.json()
-            if not isinstance(resp_body, dict) or resp_body.get("error", {}).get("code") != FACEBOOK_BATCH_ERROR_CODE:
+            if not isinstance(resp_body, dict) or (resp_body.get("error", {}).get("code") != FACEBOOK_BATCH_ERROR_CODE and resp_body.get("error", {}).get("message") != "Please reduce the amount of data you're asking for, then retry your request"):
                 # response body is not a json object or the error code is different
                 raise RuntimeError(f"Batch request failed with response: {resp_body}")
+            if resp_body.get("error", {}).get("message") == "Please reduce the amount of data you're asking for, then retry your request":
```

Comment:
ok to do in a separate PR since this is fixing a P0, but it would be useful to unit test this method